tests/api_connexion/endpoints/test_dag_run_endpoint.py

```diff
@@ -307,6 +307,41 @@ def test_should_respond_200(self, session):
             "total_entries": 2,
         }
 
+    def test_filter_by_state(self, session):
+        self._create_test_dag_run()
+        assert session.query(DagRun).count() == 2        
+        response = self.client.get(
+            "api/v1/dags/TEST_DAG_ID/dagRuns?state=running,failed", environ_overrides={'REMOTE_USER': "test"}
```

Comment:
The failed state is not tested here. Can we fail one dagrun or set the state to queued and adjust the test to match